generators/server/templates/src/main/resources/config/application.yml.ejs

```diff
@@ -217,6 +217,14 @@ spring:
               - Path=/services/<%= lowercaseBaseName %>/**
             filters:
               - RewritePath=/services/<%= lowercaseBaseName %>/?(?<segment>.*), /$\{segment}
+      <%_ for (const ms of gatewayRoutes ?? []) { _%>
+          - id: <%= ms.route %>_route
+            uri: lb://<%= ms.host %>
+            predicates:
+              - Path=/services/<%= ms.route %>/**
+            filters:
+              - RewritePath=/services/<%= ms.route %>/?(?<segment>.*), /$\{segment}
```

Comment:
It seems that the infinite redirect issue raised by @mraible might be due to the Gateway lacking a relay token, resulting in a 401 unauthorized error. After adding the TokenRelay Filter, the problem should be resolved:

```suggestion
              - RewritePath=/services/<%= ms.route %>/?(?<segment>.*), /$\{segment}
              - TokenRelay=
```